modules/tensor_mechanics/src/transfers/VectorPostprocessorTransfer.C

```diff
@@ -0,0 +1,268 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VectorPostprocessorTransfer.h"
+
+// MOOSE includes
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "MooseTypes.h"
+#include "MultiApp.h"
+#include "DelimitedFileReader.h"
+
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+#include "libmesh/parallel_algebra.h"
+
+registerMooseObject("TensorMechanicsApp", VectorPostprocessorTransfer);
+
+template <>
+InputParameters
+validParams<VectorPostprocessorTransfer>()
+{
+  InputParameters params = validParams<MultiAppTransfer>();
+  params.addClassDescription("Transfers values from a vector postprocessor in the master "
+                             "application to aux variables in the sub application.");
+  params.addRequiredParam<std::vector<AuxVariableName>>(
+      "variables",
+      "The vector of auxiliary variables to store the transferred values in master model.");
+  params.addRequiredParam<VectorPostprocessorName>(
+      "vector_postprocessor", "The vectorprocessor that contains the data in sub applications.");
+  params.addRequiredParam<std::vector<std::string>>(
+      "variable_vector_names",
+      "The names of the variable vector that need to be transferred from "
+      "the vector postprocessor to aux variables in master app.");
+  params.addParam<std::vector<SubdomainName>>(
+      "master_mesh_block_ids",
+      "If provided the vector_postprocessor values would be transferred "
+      "only to nodes belonging to those blocks.");
+  params.addParam<unsigned int>(
+      "master_component", 2, "Component along which interpolation occurs in the master model.");
+  params.addParam<unsigned int>(
+      "subapp_component", 1, "Component along which interpolation occurs in the SubApp.");
+  params.addRequiredParam<FileName>(
```

Comment:
The part about listing the positions in the other 2 components is a little confusing. Could we just input all 3 components, and have the component in the direction of the components we are interpolating along be an offset if it is nonzero?

Reply:
Yes, the offset would probably be less confusing. I will make these changes. Thanks!